### ISSUE TITLE
Cut introduction expects ground proofs

### ISSUE THREAD

**Issue by shetzl** [gapt/gapt#353]
Prover9TestCase.cut-introduction.SWV408+1 fails with
```
java.lang.IllegalArgumentException: requirement failed
	at scala.Predef$.require(Predef.scala:207)
	at at.logic.gapt.proofs.lk.cutIntroduction.Grammar.<init>(Grammar.scala:25)
	at at.logic.gapt.proofs.lk.cutIntroduction.ComputeGrammars$$anonfun$findValidGrammars$2$$anonfun$apply$8.apply(Grammar.scala:222)
	at at.logic.gapt.proofs.lk.cutIntroduction.ComputeGrammars$$anonfun$findValidGrammars$2$$anonfun$apply$8.apply(Grammar.scala:220)
	at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
	at scala.collection.immutable.List.foldLeft(List.scala:84)
	at at.logic.gapt.proofs.lk.cutIntroduction.ComputeGrammars$$anonfun$findValidGrammars$2.apply(Grammar.scala:220)
	at at.logic.gapt.proofs.lk.cutIntroduction.ComputeGrammars$$anonfun$findValidGrammars$2.apply(Grammar.scala:197)
	at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:142)
	at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:142)
	at scala.collection.immutable.HashMap$HashMap1.foreach(HashMap.scala:221)
	at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:428)
	at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:428)
	at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:142)
	at scala.collection.AbstractTraversable.foldLeft(Traversable.scala:104)
	at at.logic.gapt.proofs.lk.cutIntroduction.ComputeGrammars$.findValidGrammars(Grammar.scala:197)
	at at.logic.gapt.proofs.lk.cutIntroduction.ComputeGrammars$.findValidGrammars(Grammar.scala:113)
	at at.logic.gapt.proofs.lk.cutIntroduction.CutIntroduction$$anonfun$liftedTree1$1$1.apply(CutIntroduction.scala:277)
	at at.logic.gapt.proofs.lk.cutIntroduction.CutIntroduction$$anonfun$liftedTree1$1$1.apply(CutIntroduction.scala:246)
	at at.logic.gapt.utils.executionModels.timeout.withTimeout$$anon$1.liftedTree1$1(timeout.scala:34)
	at at.logic.gapt.utils.executionModels.timeout.withTimeout$$anon$1.run(timeout.scala:34)
```

@gebner git bisect shows that this has been introduced in commit 7a3b14fd4f09212254132d538de01d5a7ee55a6c.

**Comment by shetzl** [comment]
The same requirement fails on a number of other examples (which were not regressions from 57 to 58), e.g. GEO002-4, GEO067-2. GRP487-1, HWC003-1, LAT090-1

**Comment by gebner** [comment]
This bug happens because we get a proof from prover9 that is not ground, it contains the variables `v100`, `v102`, and `v103`; and ComputeGrammars mistakes them for non-terminals in the grammars.

We've had the same issue in the induction prover; there I just grounded all variables in the term set by replacing them with constants of the same name.  We could do the same here as well.

 * Instead of grounding term sets, we could ground the expansion sequents.
 * We could replace all variables by the same constant, possibly reducing the size of the term set.

**Comment by gebner** [comment]
Fixed by ce8e52eb459104be5ce01060c2f958c959a63829.
